homeassistant/components/vesync/switch.py

```diff
@@ -46,57 +71,46 @@ def _setup_entities(
     coordinator: VeSyncDataCoordinator,
 ):
     """Check if device is online and add entity."""
-    entities: list[VeSyncBaseSwitch] = []
-    for dev in devices:
-        if DEV_TYPE_TO_HA.get(dev.device_type) == "outlet":
-            entities.append(VeSyncSwitchHA(dev, coordinator))
-        elif DEV_TYPE_TO_HA.get(dev.device_type) == "switch":
-            entities.append(VeSyncLightSwitch(dev, coordinator))
-        else:
-            _LOGGER.warning(
-                "%s - Unknown device type - %s", dev.device_name, dev.device_type
-            )
-            continue
-
-    async_add_entities(entities, update_before_add=True)
-
+    async_add_entities(
+        (
+            VeSyncSwitchEntity(dev, description, coordinator)
+            for dev in devices
+            for description in SENSOR_DESCRIPTIONS
+            if rgetattr(dev, description.key) is not None
+        ),
+        update_before_add=True,
```

Comment:
We use a coordinator so we already have a state so we don't need to request an update before that